### ISSUE TITLE
Sub-Issue #008-1: [Model] 训练“学术润色与降重”专用 LoRA 适配器

### ISSUE THREAD

**Issue by AfterMaxQ** [AfterMaxQ/ai-word-editor#66]
*   **Parent Issue:** #008
*   **Assignee:** @YourName
*   **Status:** To Do
*   **Priority:** High
*   **Tags:** `training`, `autodl`, `dataset`, `lora`

#### 1. 背景 (Context)
通用的 LLM（如 Qwen2.5, Llama3）虽然能进行润色，但在处理“降低查重率”或“极度严谨的公文风”时，往往力度不够或风格不统一。我们需要训练一个专门的 LoRA 适配器（Adapter），挂载到本地模型上，专门响应“更学术”、“降重”等特定指令。

##### 2.1 数据集准备 (Dataset Preparation)
我们需要构建一个 `Instruction-Input-Output` 格式的数据集（Alpaca 格式），目标数量 **500 - 1000 条高质量数据**即可产生明显效果。

*   **数据策略：使用“教师模型”合成数据 (Synthetic Data)**
    *   利用 DeepSeek-V3 / GPT-4o / Claude-3.5-Sonnet 作为“教师”。
    *   构造 Prompt 让教师模型生成平行语料。

*   **数据集分类 (Categories):**
    1.  **学术润色 (Academic Polish):** 将口语化文本转换为学术被动语态、专业术语。
    2.  **降重改写 (De-duplication):** 保持原意，但大幅改变句式结构和同义词替换。
    3.  **公文规范 (Official Style):** 修正错别字，统一为简练、庄重的公文措辞。

*   **数据生成 Prompt 示例 (For Teacher Model):**
    ```markdown
    你是一个学术写作专家。请生成 20 组“草稿 -> 学术润色版”的平行语料。
    要求：
    1. 覆盖计算机科学、管理学、马克思主义哲学等常见论文领域。
    2. "Input" 是学生写的口语化草稿。
    3. "Output" 是符合 SCI/核心期刊标准的表达。
    4. 输出为 JSON 格式。
    ```

*   **目标数据格式 (`dataset_polish.json`):**
    ```json
    [
      {
        "instruction": "请将这段文本修改为学术风格，用于论文摘要。",
        "input": "我们做了一个实验，发现这个算法比以前的快了很多，而且更准了。",
        "output": "实验结果表明，该算法在计算效率和准确性方面均显著优于现有基准方法。"
      },
      {
        "instruction": "对以下文本进行降重处理，保持原意但改变表述。",
        "input": "人工智能技术现在发展很快，很多行业都开始用它来提高效率。",
        "output": "随着人工智能技术的飞速演进，众多产业正通过集成该技术以实现效能的飞跃。"
      }
    ]
    ```

- [ ] 生成/收集 1000 条平行语料。
- [ ] 清洗数据，确保 JSON 格式无误。

##### 2.2 AutoDL 环境配置 (Environment Setup)
*   **推荐镜像:** PyTorch 2.1.2 + CUDA 12.1 + Python 3.10
*   **推荐显卡:** RTX 4090 (24GB) 或 A5000 (24GB) —— 训练 7B/14B 模型性价比最高。
*   **训练框架:** 推荐使用 **LLaMA-Factory** (WebUI 方便调整) 或 **Unsloth** (速度极快，省显存)。

*   **AutoDL 初始化脚本示例:**
    ```bash
    # 如果使用 LLaMA-Factory
    git clone --depth 1 https://github.com/hiyouga/LLaMA-Factory.git
    cd LLaMA-Factory
    pip install -e ".[torch,metrics]"
    ```

##### 2.3 训练参数配置 (Training Configuration)
以基座模型 **Qwen2.5-7B-Instruct** 为例（适合中文写作，逻辑强）：

*   **Stage:** `sft` (Supervised Fine-Tuning)
*   **Quantization:** `4bit` (QLoRA，为了在单卡上跑起来)
*   **LoRA Config:**
    *   `lora_rank`: 32 或 64
    *   `lora_alpha`: 16
    *   `lora_dropout`: 0.05
    *   `target_modules`: `all-linear` (效果最好)
*   **Hyperparameters:**
    *   `num_train_epochs`: 3 - 5 (数据少，多跑几轮，注意观察 Loss)
    *   `learning_rate`: 2e-4 (QLoRA 常用)
    *   `per_device_train_batch_size`: 4 (根据显存调整)
    *   `gradient_accumulation_steps`: 4

- [ ] 上传基座模型 (或直接在 AutoDL 用 modelscope 下载)。
- [ ] 上传数据集 `dataset_polish.json`。
- [ ] 启动训练并监控 Loss 曲线 (应下降至 0.5-0.8 左右)。

##### 2.4 导出与转换 (Export & Conversion)
训练完成后，需要将 LoRA 适配器转换为 GGUF 格式，以便你本地的 `llama.cpp` 使用。

1.  **合并导出 (Merge LoRA):** 在 AutoDL 上将 LoRA 权重合并回基座模型（得到完整的 FP16 模型）。
2.  **GGUF 转换:**
    *   使用 `llama.cpp` 自带的 `convert_hf_to_gguf.py`。
    *   或者，如果不合并，直接将 LoRA adapter 转换为 GGUF 格式（llama.cpp 支持挂载独立的 LoRA 文件 `--lora`）。
    *   *建议：* 为了方便分发和管理，建议**不合并**，只转换 LoRA Adapter 为 GGUF，这样用户只需要下载几百 MB 的 Adapter，挂载到通用的 Qwen2.5 GGUF 上即可。

    ```bash
    # 在 AutoDL 上使用 llama.cpp 脚本转换 LoRA
    python convert_lora_to_gguf.py /path/to/your/saved/lora/folder --outtype f16 --outfile academic_polish_lora_f16.gguf
    ```

##### 2.5 本地集成测试 (Integration)
将生成的 `academic_polish_lora_f16.gguf` 复制到你本地的 `G:\Python\ai-word-editor\models\`。

*   **测试命令:**
    ```powershell
    # 假设你用的是 main.exe 或 server.exe
    ./llama-server.exe -m models/qwen2.5-7b-instruct-q5_k_m.gguf --lora models/academic_polish_lora_f16.gguf --lora-scaled 1.0
    ```
*   **前端调用:**
    在发起润色请求时，Prompt 必须包含你在训练数据集中使用的关键词（如“学术风格”、“降重”），以激活 LoRA 的能力。

#### 3. 验收标准 (Definition of Done)
1.  [ ] 成功产出 `academic_polish_lora.gguf` 文件。
2.  [ ] 本地加载 LoRA 后，输入“把这句话改成学术风格：[口语]”，输出结果明显优于不挂载 LoRA 的基座模型。
3.  [ ] 降重指令能有效改变句式结构，且不改变原意。
